docs/documentation/siddhi-4.0.md

```diff
@@ -2163,4 +2163,98 @@ _We value your contribution on improving Siddhi and its extensions further._
 
 Custom extensions can be written in order to cater use case specific logic that are not available in Siddhi out of the box or as an existing extension. 
 
-More information on this will be available soon.
+There are five types of Siddhi extensions that you can write to cater your specific use cases. Please find each 
+extension types and related maven archetypes below. You can use these archetypes to generate maven projects for each 
+extension type.
+
+* Follow one of the step below, based on your project :
+
+
+**siddhi-execution**
+
```

Comment:
Tell what type of extensions can be written using execution type, explain about that a bit more.

Reply:
Added brief description for each archetype.